test/test_tables.py

```diff
@@ -305,15 +305,264 @@ class DoublyNested(qv.Table):
+
+    sorted3 = pair.sort_by([("x", "ascending")])
+    assert sorted3.x.to_pylist() == [1, 2, 3]
+    assert sorted3.y.to_pylist() == [5, 1, 2]
+
+    sorted4 = pair.sort_by([("x", "descending")])
+    assert sorted4.x.to_pylist() == [3, 2, 1]
+    assert sorted4.y.to_pylist() == [2, 1, 5]
+
+
+def test_sort_by_with_column_names():
+    pair = Pair.from_kwargs(x=[1, 2, 3], y=[5, 1, 2])
+
     sorted1 = pair.sort_by("y")
-    assert sorted1.x[0].as_py() == 2
-    assert sorted1.x[1].as_py() == 3
-    assert sorted1.x[2].as_py() == 1
-
-    sorted2 = pair.sort_by([("x", "descending")])
-    assert sorted2.x[0].as_py() == 3
-    assert sorted2.x[1].as_py() == 2
-    assert sorted2.x[2].as_py() == 1
+    assert sorted1.x.to_pylist() == [2, 3, 1]
+    assert sorted1.y.to_pylist() == [1, 2, 5]
+
+    sorted2 = pair.sort_by("x")
+    assert sorted2.x.to_pylist() == [1, 2, 3]
+    assert sorted2.y.to_pylist() == [5, 1, 2]
+
+
+def test_sort_by_multiple_columns():
+    pair = Pair.from_kwargs(x=[2, 1, 2, 3], y=[6, 4, 5, 6])
+
+    # Column order: y first then x (alternating ascending/descending)
+    sorted1 = pair.sort_by([("y", "ascending"), ("x", "ascending")])
+    assert sorted1.x.to_pylist() == [1, 2, 2, 3]
+    assert sorted1.y.to_pylist() == [4, 5, 6, 6]
+
+    sorted2 = pair.sort_by([("y", "ascending"), ("x", "descending")])
+    assert sorted2.x.to_pylist() == [1, 2, 3, 2]
+    assert sorted2.y.to_pylist() == [4, 5, 6, 6]
+
+    sorted3 = pair.sort_by([("y", "descending"), ("x", "ascending")])
+    assert sorted3.x.to_pylist() == [2, 3, 2, 1]
+    assert sorted3.y.to_pylist() == [6, 6, 5, 4]
+
+    sorted4 = pair.sort_by([("y", "descending"), ("x", "descending")])
+    assert sorted4.x.to_pylist() == [3, 2, 2, 1]
+    assert sorted4.y.to_pylist() == [6, 6, 5, 4]
+
+    # Column order: x first then y (alternating ascending/descending)
+    sorted5 = pair.sort_by([("x", "ascending"), ("y", "ascending")])
+    assert sorted5.x.to_pylist() == [1, 2, 2, 3]
+    assert sorted5.y.to_pylist() == [4, 5, 6, 6]
+
+    sorted6 = pair.sort_by([("x", "ascending"), ("y", "descending")])
+    assert sorted6.x.to_pylist() == [1, 2, 2, 3]
+    assert sorted6.y.to_pylist() == [4, 6, 5, 6]
+
+    sorted7 = pair.sort_by([("x", "descending"), ("y", "ascending")])
+    assert sorted7.x.to_pylist() == [3, 2, 2, 1]
+    assert sorted7.y.to_pylist() == [6, 5, 6, 4]
+
+    sorted8 = pair.sort_by([("x", "descending"), ("y", "descending")])
+    assert sorted8.x.to_pylist() == [3, 2, 2, 1]
+    assert sorted8.y.to_pylist() == [6, 6, 5, 4]
+
+
+def test_sort_by_multiple_columns_with_column_names():
+    pair = Pair.from_kwargs(x=[2, 1, 2, 3], y=[6, 4, 5, 6])
+
+    # Column order: y first then x
+    sorted1 = pair.sort_by(["y", "x"])
+    assert sorted1.x.to_pylist() == [1, 2, 2, 3]
+    assert sorted1.y.to_pylist() == [4, 5, 6, 6]
+
+    sorted2 = pair.sort_by(["x", "y"])
+    assert sorted2.x.to_pylist() == [1, 2, 2, 3]
+    assert sorted2.y.to_pylist() == [4, 5, 6, 6]
```

Comment:
> - Passing a list of column names instead of a list of tuples

I added support for this but happy to remove it if you think its not necessary.

Reply:
Ah, I meant that it should raise an error. But as soon as I see it, I like it - it's nice to simplify this for the common case!